src/components/Courses/tests/courses.test.js

```diff
@@ -0,0 +1,85 @@
+import { fireEvent, render, screen } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import { Switch } from 'react-router-dom';
+import { BrowserRouter as Router } from 'react-router-dom/cjs/react-router-dom.min';
+import { mockedAuthorsList, mockedCoursesList } from '../../../constants';
+import CourseForm from '../../CourseForm/CourseForm';
+import Courses from '../Courses';
+
+const mockedState_C_A = {
+	user: {
+		isAuth: true,
+		name: 'Test Name',
+		token: 'test',
+		role: 'admin',
+	},
+	courses: mockedCoursesList,
+	authors: mockedAuthorsList,
+};
+
+const mockedStore_C_A = {
+	getState: () => mockedState_C_A,
+	subscribe: jest.fn(),
+	dispatch: jest.fn(),
+};
+
+const mockedState_NoC_A = {
+	user: {
+		isAuth: true,
+		name: 'Test Name',
+		token: 'test',
+		role: 'admin',
+	},
+	courses: [],
+	authors: mockedAuthorsList,
+};
+
+const mockedStore_NoC_A = {
+	getState: () => mockedState_NoC_A,
+	subscribe: jest.fn(),
+	dispatch: jest.fn(),
+};
+
```

Comment:
You can wrap all your tests with Describe https://jestjs.io/docs/api#describename-fn.  This is a good practice because it groups your tests. And also if e.g. you want to run all tests manually, you will be able to do it clicking on describe 
![image](https://user-images.githubusercontent.com/28926420/156551857-dcf60614-ed3d-416b-9aaa-c3bbeb7738df.png)

Reply:
Done! (also the jest extension for IDE is very useful)